src/test/stateful_modelinstance_test.cpp

```diff
@@ -264,37 +285,92 @@ TEST_F(StatefulModelInstance, PreprocessingFirstRequest) {
+        const ovms::model_memory_state_t& irMemoryState = inferRequest.QueryState();
+        EXPECT_EQ(irMemoryState.size(), 1);
+        EXPECT_EQ(irMemoryState[0].GetName(), "state");
 
-        // Prepare new state blob
-        std::vector<float> newState(elementsCount);
-        std::iota(newState.begin(), newState.end(), 10);
-        Blob::Ptr newBlob = make_shared_blob<float>(desc, newState.data());
+        InferenceEngine::Blob::Ptr stateCloneBlob = nullptr;
+        EXPECT_EQ(ovms::blobClone(stateCloneBlob, irMemoryState[0].GetState()), ovms::StatusCode::OK);
+
+        std::vector<float> currentBlobIrData;
+        currentBlobIrData.assign((float*)stateCloneBlob->buffer(), ((float*)stateCloneBlob->buffer()) + elementsCount);
+        EXPECT_EQ(currentBlobIrData, defaultState);
+
+        // Inject sequence with newState as the last state written to sequence memory state
+        ovms::model_memory_state_t memoryState;
+        addState(memoryState, "state", shape, newState);
+        modelInstance->injectSequence(sequenceId, memoryState);
+
+        // Perform preprocessing (load state from sequence to infer request)
+        modelInstance->preInferenceProcessing(inferRequest, sequenceProcessingSpec);
+
+        // Check if InferRequest memory state has been updated to sequence memory state
+        EXPECT_EQ(ovms::blobClone(stateCloneBlob, irMemoryState[0].GetState()), ovms::StatusCode::OK);
+        currentBlobIrData.assign((float*)stateCloneBlob->buffer(), ((float*)stateCloneBlob->buffer()) + elementsCount);
+        EXPECT_EQ(currentBlobIrData, newState);
+    }
+}
+
+TEST_F(StatefulModelInstanceTest, PostprocessingLastRequest) {
+    // Prepare model instance and processing spec
+    uint32_t sequenceControlInput = SEQUENCE_END;
+    uint64_t sequenceId = 42;
+    ovms::SequenceProcessingSpec sequenceProcessingSpec(sequenceControlInput, sequenceId);
+
+    // Initialize InferRequest with current state and default state
+    std::shared_ptr<IInferRequest> iireqPtr = std::make_shared<MockIInferRequestStateful>("state", currentBlob, defaultBlob);
+    InferRequest inferRequest(iireqPtr);
+
+    // Check if InferRequest has been initialized properly
+    const ovms::model_memory_state_t& irMemoryState = inferRequest.QueryState();
+    EXPECT_EQ(irMemoryState.size(), 1);
+    EXPECT_EQ(irMemoryState[0].GetName(), "state");
+
+    InferenceEngine::Blob::Ptr stateCloneBlob = nullptr;
+    EXPECT_EQ(ovms::blobClone(stateCloneBlob, irMemoryState[0].GetState()), ovms::StatusCode::OK);
+
+    std::vector<float> currentBlobIrData;
+    currentBlobIrData.assign((float*)stateCloneBlob->buffer(), ((float*)stateCloneBlob->buffer()) + elementsCount);
+    EXPECT_EQ(currentBlobIrData, currentState);
+
+    tensorflow::serving::PredictResponse response;
+
+    modelInstance->postInferenceProcessing(&response, inferRequest, sequenceProcessingSpec);
+
+    auto& output = (*response.mutable_outputs())["sequence_id"];
```

Comment:
This might be confusing because if response does not contain `sequence_id` entry, it will create one.
I think we should check if (*response.mutable_outputs()) contains `sequence_id` beforehand.